src/main/daml/Daml/Finance/Interface/Instrument/Bond/Callable/BondTypes.daml

```diff
@@ -4,20 +4,38 @@
 module Daml.Finance.Interface.Instrument.Bond.Callable.BondTypes where
 
 import Daml.Finance.Interface.Types.Date.Calendar (BusinessDayConventionEnum)
+import Daml.Finance.Interface.Types.Date.DayCount (DayCountConventionEnum)
 import Daml.Finance.Interface.Types.Date.RollConvention (PeriodEnum)
 
 -- | Specifies the data required for a floating rate coupon.
 data FloatingRate = FloatingRate
   with
     referenceRateId : Text
       -- ^ The identifier of the reference rate to be used for the coupon, e.g. Libor-3M.
-    resetRelativeTo : DateRelativeToEnum
-      -- ^ Indicates whether the reference rate will reset relative to the first or the last day of
-      --   the calculation period.
+    referenceRateType : ReferenceRateTypeEnum
+      -- ^ The type of reference rate, which defines how the reference rate is calcuated.
     fixingDates : FixingDates
       -- ^ Specifies the fixing dates as an offset of the calculation date, e.g. -2 business days.
   deriving (Eq, Show)
 
+-- | The type of reference rate, which defines how the reference rate is calcuated.
+data ReferenceRateTypeEnum
```

Comment:
Are we copy-pasting Fpml types here? If so, it might make sense to avoid duplication

Reply:
Only two types: `DateRelativeToEnum` and `FixingDates`. The others are adapted for bonds. Do you think it is worth reusing them? If so, we should probably put them in a common package, to avoid a dependency on `Swap` in `Bond`.

Reply:
if it's just two data types then no need, but I would raise an issue to track this.

I wonder how it works when building a bond or a swap using e.g. the Java bindings, whether we can re-use the same logic although the data types belong to two different packages

Reply:
Also, I would keep track of types that get adapted for bonds, as we might need to realign them at some point

Reply:
I opened an issue for this: https://github.com/digital-asset/daml-finance/issues/791